simtools/simulator.py

```diff
@@ -313,42 +328,55 @@ def _set_simulation_runner(self):
         Set simulation runners
 
         """
+        common_args = {
+            "label": self.label,
+            "simtel_source_path": self._simulator_source_path,
+        }
+        corsika_args = {
+            "mongo_db_config": self._mongo_db_config,
+            "site": self.site,
+            "layout_name": self.layout_name,
+            "corsika_parameters_file": self._corsika_parameters_file,
+            "corsika_config_data": self._corsika_config_data,
+        }
+        # TODO: This is not very elegant, find a nicer solution?
+        if self.simulator in ["simtel", "corsika_simtel"]:
```

Comment:
Explain to me what the issue is here.

Reply:
I am not happy about the fact there are two very similar `if`s one after the other:

```
 # TODO: This is not very elegant, find a nicer solution?
        if self.simulator in ["simtel", "corsika_simtel"]:
            simtel_args = {
                "array_model": self.array_model,
                "config_data": {
                    "simtel_data_directory": self.config.data_directory,
                    "primary": self.config.primary,
                    "zenith_angle": self.config.zenith_angle * u.deg,
                    "azimuth_angle": self.config.azimuth_angle * u.deg,
                },
            }
        if self.simulator == "corsika":
            self._set_corsika_runner(common_args | corsika_args)
        if self.simulator == "simtel":
            self._set_simtel_runner(common_args | simtel_args)
        if self.simulator == "corsika_simtel":
            self._set_corsika_simtel_runner(common_args, corsika_args, simtel_args)
```

This is just because the code crashes otherwise in case we set `corsika` as the simulator and `self.config` is `None`. I would have liked to avoid this first if and somehow call `config.data_directory` without crashing. Perhaps if I change it to a dictionary or call it like a dictionary it would work. I would like to try anyway.

Reply:
OK, understand. I don't have a better solution to suggest.